Fix text bleeding in EPUB reader column layouts (#4040)

MAVRICK-1 opened the pull request:
This pull request updates the page size and layout calculations in the `BookReaderComponent` to improve accuracy and consistency for different page layout modes and calculation methods. The changes focus on simplifying the logic and correcting the computed values for column layouts.

**Layout Calculation Updates:**

* The logic for calculating the page width in `BookPageLayoutMode.Column1` and `BookPageLayoutMode.Column2` has been simplified to use direct division of the `base` value, removing extra offsets and unnecessary client width checks.

**Page Size Calculation Improvements:**

* The page size calculation now consistently uses a new `baseWidth` variable and subtracts the column gap instead of adding it, which corrects the computed page size for both calculation methods.

**Column Modifier Adjustment:**

* The column modifier for `BookPageLayoutMode.Column2` has been updated to return `2` or `2.5` (instead of `1` or `1.25`), ensuring the correct number of columns is used in the layout calculation logic.


Fixes #4040 
/claim #4040

MAVRICK-1 committed:
Fix text bleeding in EPUB reader column layouts (#4040)

- Fix pageWidth calculation: subtract column gaps instead of adding them
- Increase columnGapModifier values for Column2 mode (2/2.5 vs 1/1.25)
- Unify columnWidth calculations removing magic numbers
- Resolves text bleeding during page navigation in 1 and 2 Column layout modes

majora2007 commented on UI/Web/src/app/book-reader/_components/book-reader/book-reader.component.ts:
```diff
@@ -1714,7 +1713,7 @@ export class BookReaderComponent implements OnInit, AfterViewInit, OnDestroy {
         return 1;
       case BookPageLayoutMode.Column2:
-        return calculationMethod === EpubPageCalculationMethod.Default ? 1 : 1.25;
+        return calculationMethod === EpubPageCalculationMethod.Default ? 2 : 2.5;
```
The goal of the fix is to remove any need of having 2 different calculation modes.

majora2007 commented:
@MAVRICK-1 can you remove the 2 calculation modes and only use one? Then I can deploy this out to our userbase and get it tested out.

MAVRICK-1 commented:
https://github.com/user-attachments/assets/2e53b7a4-e3a0-4901-9211-a851e228ee32


@majora2007 needed help on this the book scan is always in queued , the books are not loading . i have started tha api server and frontend also but the scanning dont work

majora2007 commented:
Make sure your files are not just at the root, so if your library folder is `C:/myfiles/`, your files need to be at least in one folder `C:/myfiles/files/`. 

https://wiki.kavitareader.com/guides/scanner/ has documentation that can help you. Otherwise, shoot me over your logs.

MAVRICK-1 commented:
<img width="1920" height="1200" alt="image" src="https://github.com/user-attachments/assets/af97f6a1-31e5-4528-8cb0-ff3a2e157ca4" />


cc @majora2007  will push few more changes

MAVRICK-1 committed:
Simplify EPUB text bleeding fix - remove dual calculation modes

- Remove EpubPageCalculationMethod checks from pageWidth() function
- Remove calculationMethod dependency from columnGapModifier()
- Use single unified calculation: baseWidth - (COLUMN_GAP * columnGapModifier)
- Set Column2 gap modifier to 2 (removing 2.5 variant)
- Addresses text bleeding in Column1/Column2 layouts during page navigation

Fixes #4040

majora2007 commented:
Please make sure you also test against the Japanese book (change writing style to vertical) and try different sizes of the viewport to validate before handoff. 

Once you're done, ping me and ill get this out to our dedicated testers and give you any feedback.

MAVRICK-1 committed:
Fix text bleeding for both horizontal and vertical writing modes

- Update getVerticalPageWidth() to account for column gaps
- Now applies COLUMN_GAP * columnGapModifier for vertical layouts
- Fixes Japanese/vertical text bleeding in Column1/Column2 modes
- Ensures consistent gap handling across all writing styles
- Works with all viewport sizes and text orientations

This ensures the fix works for both:
- English/horizontal text (pageWidth function)
- Japanese/vertical text (getVerticalPageWidth function)

Fixes #4040

MAVRICK-1 committed:
Fix column width calculations to prevent text bleeding

- Column1: Use (base - COLUMN_GAP) instead of (base / 2)
- Column2: Use ((base - COLUMN_GAP) / 2) instead of (base / 4)
- Prevents browser from creating extra columns beyond intended count
- Ensures column-width and column-count work together properly
- Supports all viewport sizes by using proper width calculations

This resolves the remaining text bleeding issues seen in the screenshots
where text was overflowing beyond column boundaries.

Fixes #4040

MAVRICK-1 committed:
Fix CSS column-gap conflict causing text bleeding

- Set CSS column-gap to 0px in column-layout-1 and column-layout-2
- Remove COLUMN_GAP from columnWidth() calculations
- Use simpler column width: base for Column1, base/2 for Column2
- Eliminates conflict between CSS column-gap and JS column-width
- Prevents browser from creating extra unwanted columns

This resolves the text bleeding issue by ensuring only the intended
number of columns are created with proper widths.

Fixes #4040

majora2007 commented:
I'm going to close this attempt out. I haven't heard from you. Feel free to reopen the PR and tag me if you make progress.
